terraform/java/ec2/asg/main.tf

```diff
@@ -152,6 +152,7 @@ resource "aws_autoscaling_group" "asg" {
   max_size = 1
   launch_configuration = aws_launch_configuration.launch_configuration.name
   vpc_zone_identifier = [data.aws_subnets.default_subnets.ids.0]
+  health_check_type = "EC2"
```

Comment:
What does this do?

Reply:
Tells the ASG to check if EC2 instances are not up, and if not then terminate and create new one

Reply:
Does that create a loop if there is a mistake in the code that would lead to the ec2 instance never coming up? Does the terraform behave normally in this case?

I would recommend trying to put a syntax error in the launch code that starts the instrumentation and sample app and seeing what happens

Reply:
Yeah already tried that. Unfortunately code in the launch template doesn't cause the instance to fail to start running